packages/typespec-azure-resource-manager/test/resource-resolution.test.ts

```diff
@@ -257,39 +258,42 @@ describe("unit tests for resource manager helpers", () => {
         title: "generic extension resource weird read path with default",
         path: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{providerName}/{resourceType}/{resourceName}/{childResourceType}/{childResourceName}/providers/Microsoft.Bar/bars/{barName}/basses/default/actionName/doSomething/doSomethingElse/andAnotherThing",
         kind: "read",
+        singletonResourceName: "default",
         expected: {
           resourceType: {
             provider: "Microsoft.Bar",
-            types: ["bars", "basses"],
+            types: ["bars"],
           },
           resourceInstancePath:
-            "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{providerName}/{resourceType}/{resourceName}/{childResourceType}/{childResourceName}/providers/Microsoft.Bar/bars/{barName}/basses/default",
+            "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{providerName}/{resourceType}/{resourceName}/{childResourceType}/{childResourceName}/providers/Microsoft.Bar/bars/{barName}",
```

Comment:
the result changed because of [the change above](https://github.com/Azure/typespec-azure/pull/3594/files#r2591600872).
In this path, we have a `default` but it is not at the end.

Also I have concerns: why would we try to rebuild the resourceInstancePath from the path any way? Should we just use the READ request path as the resource instance path? I think ARM did this in their logic, and resource type is then built out of this path.